src/models/campaign.ts

```diff
@@ -138,9 +139,11 @@ export default class Campaign {
     /* Save */
 
     public async save(): Promise<Response<string>> {
+        const dashboardId = await this.db.createDashboard(this.name);
         const teamsCode = metadataConfig.categoryComboCodeForTeams;
         const antigenCodes = this.antigens.map(antigen => antigen.code);
         const vaccinationAttribute = await this.db.getAttributeIdByCode(metadataConfig.attibuteCodeForApp);
+        const dashboardAttribute = await this.db.getAttributeIdByCode(metadataConfig.attributeCodeForDashboard);
```

Comment:
We discussed the other day when to use "getOne" or "getMany" functions. Here we are doing now two requests to get attributes, it could be done in one with a `ByCodes` function. Not necessary to change it now, just as a note. The parent code is slightly more complex since you have to index the response and get the values for each attribute (order in the response is unknown). There is an endpoint I am starting to use in another branch, GET /metadata ([docs](https://docs.dhis2.org/2.30/en/developer/html/dhis2_developer_manual_full.html#webapi_metadata_export)), it will be handy to get all the metadata we need in a single request.

Reply:
As discussed, @tokland will take care of this as part of the disaggregation PR.